src/Core/Layer/Collector/ComposerFilesParser.php

```diff
@@ -71,12 +72,18 @@ public function __construct(string $lockFile)
      * @param string[] $requirements
      *
      * @return string[]
+     *
+     * @throws InvalidCollectorDefinitionException
      */
     public function autoloadableNamespacesForRequirements(array $requirements, bool $includeDev): array
     {
         $namespaces = [[]];
 
         foreach ($requirements as $package) {
+            if (!array_key_exists($package, $this->lockedPackages)) {
+                throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('ComposerCollector got a package that doesn\'t exist: "%s"', $package));
```

Comment:
You are referencing `ComposerCollector` inside this class `ComposerFilesParser`. The parser should be oblivious to how/by whom it is used. 

You could instead just throw a generic `RuntimeException` and rethrow it as `InvalidCollectorDefinitionException` in the `ComposerCollector.`